include/ddc/kernels/splines/splines_linear_problem_2x2_blocks.hpp

```diff
@@ -109,6 +144,59 @@ class SplinesLinearProblem2x2Blocks : public SplinesLinearProblem<ExecSpace>
         }
     }
 
+    /**
+     * @brief Fill a COO version of a Dense matrix (remove zeros).
+     *
+     * [SHOULD BE PRIVATE (GPU programming limitation)]
+     *
+     * Runs on a single thread to garantee ordering.
+     *
+     * @param dense_matrix The dense storage matrix whose non-zeros are extracted to fill the COO matrix.
+     *
+     * @return The COO storage matrix filled with the non-zeros from dense_matrix.
+     */
+    Coo dense2coo(Kokkos::View<double**, Kokkos::LayoutRight, typename ExecSpace::memory_space>
+                          dense_matrix)
+    {
+        Kokkos::View<int*, Kokkos::LayoutRight, typename ExecSpace::memory_space> rows_idx(
+                "ddc_splines_coo_rows_idx",
+                dense_matrix.extent(0) * dense_matrix.extent(1));
+        Kokkos::View<int*, Kokkos::LayoutRight, typename ExecSpace::memory_space> cols_idx(
+                "ddc_splines_coo_cols_idx",
+                dense_matrix.extent(0) * dense_matrix.extent(1));
+        Kokkos::View<double*, Kokkos::LayoutRight, typename ExecSpace::memory_space>
+                values("ddc_splines_coo_values", dense_matrix.extent(0) * dense_matrix.extent(1));
+
+        Kokkos::DualView<std::size_t, Kokkos::LayoutRight, typename ExecSpace::memory_space>
+                n_nonzeros("ddc_splines_n_nonzeros");
+        n_nonzeros.h_view() = 0;
+        n_nonzeros.modify_host();
+        n_nonzeros.sync_device();
+
+        Kokkos::parallel_for(
+                "dense2coo",
+                Kokkos::RangePolicy(ExecSpace(), 0, 1),
+                KOKKOS_LAMBDA(const int) {
+                    for (int i = 0; i < dense_matrix.extent(0); i++) {
+                        for (int j = 0; j < dense_matrix.extent(1); j++) {
+                            double aij = dense_matrix(i, j);
+                            if (Kokkos::abs(aij) >= 1e-14) {
+                                rows_idx(n_nonzeros.d_view()) = i;
+                                cols_idx(n_nonzeros.d_view()) = j;
+                                values(n_nonzeros.d_view()++) = aij;
```

Comment:
```suggestion
                                rows_idx(n_nonzeros.d_view()) = i;
                                cols_idx(n_nonzeros.d_view()) = j;
                                values(n_nonzeros.d_view()) = aij;
                                n_nonzeros.d_view()++;
```

Let's avoid too complicated lines